From 6b55d07dc2e23752e417ebbc3dfcd649a4508395 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Thu, 3 Apr 2014 10:53:34 +0100
Subject: Avoid bashisms

"command &> /dev/null" is a bashism and will fail on e.g. systems where
/bin/sh is dash.  Use a POSIX-compatible alternative.

Forwarded: yes
Last-Update: 2014-04-03

Patch-Name: avoid-bashisms.patch
---
 iprlib.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/iprlib.c b/iprlib.c
index 33f8769..49e11e5 100644
--- a/iprlib.c
+++ b/iprlib.c
@@ -3544,19 +3544,19 @@ int ipr_format_unit(struct ipr_dev *dev)
 	struct ipr_dev *multipath_dev;
 
 	if (strlen(dev->dev_name) && dev->scsi_dev_data->device_id) {
-		sprintf(cmnd, "/sbin/multipathd -k\"del path %s\" &> /dev/null", strrchr(dev->dev_name, '/') + 1);
+		sprintf(cmnd, "/sbin/multipathd -k\"del path %s\" > /dev/null 2>&1", strrchr(dev->dev_name, '/') + 1);
 		system(cmnd);
 		sprintf(cmnd, "/bin/rm -rf %s %s%s", dev->dev_name, dev->dev_name, "[0-9]*");
 		system(cmnd);
 
 		multipath_dev = find_multipath_jbod(dev);
 		if (multipath_dev) {
-			sprintf(cmnd, "/sbin/multipathd -k\"del path %s\" &> /dev/null", strrchr(multipath_dev->dev_name, '/') + 1);
+			sprintf(cmnd, "/sbin/multipathd -k\"del path %s\" > /dev/null 2>&1", strrchr(multipath_dev->dev_name, '/') + 1);
 			system(cmnd);
 			sprintf(cmnd, "/bin/rm -rf %s %s%s", multipath_dev->dev_name, multipath_dev->dev_name , "[0-9]*");
 			system(cmnd);
 		}
-		system("/sbin/multipath -F &> /dev/null");
+		system("/sbin/multipath -F > /dev/null 2>&1");
 	}
 
 	if (strlen(name) == 0)
